crates/lib/examples/execute.rs

```diff
@@ -19,39 +22,47 @@ async fn quilc_client() -> rpcq::Client {
 
 #[tokio::main]
 async fn main() {
-    let mut exe = Executable::from_quil(PROGRAM).with_quilc_client(Some(quilc_client().await));
+    let mut exe = Executable::from_quil(PROGRAM); //.with_quilc_client(Some(quilc_client().await));
     let execution_options = ExecutionOptions::default();
 
     // You can submit a job to a QPU and retrieve results in a single step.
-    let result = exe
-        .with_parameter("theta", 0, PI)
+    exe.with_parameter("theta", 0, PI)
+        .with_parameter("ro", 0, 0f64)
+        .with_parameter("ro", 1, 0f64)
+        .with_parameter("some_int", 0, 0f64);
+
+    let mut executable = exe.with_shots(10.try_into().unwrap());
+
+    let result = executable
         .execute_on_qpu("Ankaa-2", None, &execution_options)
         .await
         .expect("Program should execute successfully");
 
     dbg!(&result);
 
     // Or you can submit a job, then retrieve results later.
-    let handle = exe
-        .with_parameter("theta", 0, PI / 2.0)
-        .submit_to_qpu("Ankaa-2", None, &execution_options)
-        .await
-        .expect("Program should be sumbitted successfully.");
+    // let handle = exe
+    //     .with_parameter("theta", 0, PI / 2.0)
+    //     .with_parameter("ro", 0, 0f64)
+    //     .with_parameter("ro", 1, 0f64)
+    //     .submit_to_qpu("Ankaa-2", None, &execution_options)
+    //     .await
+    //     .expect("Program should be sumbitted successfully.");
 
     // You can use the job handle to attempt job cancellation. This will only succeed if the job
     // has not begun executing.
-    let cancelled_result = exe.cancel_qpu_job(handle.clone()).await;
-    dbg!(&cancelled_result);
-    let cancelled = cancelled_result.is_ok();
+    // let cancelled_result = exe.cancel_qpu_job(handle.clone()).await;
+    // dbg!(&cancelled_result);
+    // let cancelled = cancelled_result.is_ok();
 
     // Retrieving results will return an error if the job was successfully cancelled.
-    let result = exe.retrieve_results(handle.clone()).await;
-    if !cancelled {
-        println!("Job was not cancelled");
-        assert!(result.is_ok());
-    } else {
-        println!("Job was cancelled");
-        assert!(result.is_err());
-    }
-    dbg!(&result);
+    // let result = exe.retrieve_results(handle.clone()).await;
+    // if !cancelled {
+    //     println!("Job was not cancelled");
+    //     assert!(result.is_ok());
+    // } else {
+    //     println!("Job was cancelled");
+    //     assert!(result.is_err());
+    // }
+    // dbg!(&result);
```

Comment:
Should all of this remain commented too?

Reply:
whoops, nope, removed